website/addons/dropbox/views/auth.py

```diff
@@ -151,25 +150,32 @@ def dropbox_user_config_get(user_addon, auth, **kwargs):
         'create': api_url_for('dropbox_oauth_start_user'),
         'delete': api_url_for('dropbox_oauth_delete_user')
     }
-    info = user_addon.dropbox_info
-    valid_credentials = True
-
-    if user_addon.has_auth:
-        try:
-            client = get_client_from_user_settings(user_addon)
-            client.account_info()
-        except ErrorResponse as error:
-            if error.status == 401:
-                valid_credentials = False
-            else:
-                HTTPError(http.BAD_REQUEST)
+    user_addon = auth.user.get_addon('dropbox')
 
+    info = None
+    user_has_auth = False
+    n_nodes_authorized = 0
+    valid_credentials = True
+    if user_addon:
+        info = user_addon.dropbox_info
+        user_has_auth = user_addon.has_auth
+        n_nodes_authorized = len(user_addon.nodes_authorized)
+
+        if user_addon.has_auth:
+            try:
+                client = get_client_from_user_settings(user_addon)
+                client.account_info()
+            except ErrorResponse as error:
+                if error.status == 401:
+                    valid_credentials = False
+                else:
+                    raise HTTPError(http.BAD_REQUEST)
```

Comment:
Might be good to add a `message_long` here; can we get any info from the `ErrorResponse`?